src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -67,5 +67,12 @@
    :http-client-idle-timeout-milliseconds schema/Int
    :http-client-metrics-enabled schema/Bool
    :max-requests-per-instance schema/Int
-   (schema/optional-key :boltlib-path) [schema/Str]})
+   (schema/optional-key :boltlib-path) [schema/Str]
+   ;; Deprecated in favor of their `server-*` counterparts.
+   ;; to be removed in Puppet 8.
+   (schema/optional-key :master-conf-dir) (schema/maybe schema/Str)
+   (schema/optional-key :master-code-dir) (schema/maybe schema/Str)
+   (schema/optional-key :master-var-dir) (schema/maybe schema/Str)
+   (schema/optional-key :master-run-dir) (schema/maybe schema/Str)
+   (schema/optional-key :master-log-dir) (schema/maybe schema/Str)})
```

Comment:
I guess we lose a little bit here, in that it seems that now we'd allow neither variant to be specified, but I'm not really worried about it.

Reply:
What would happen if neither were specified?

Reply:
I don't think it's technically possible unless we messed something up in code, since we fill in defaults for everything.

Reply:
ahkay, so a user doesn't have to worry about making sure one of these is specified? then yeah, I'm not really worried about it either.